src/librustc/middle/resolve_lifetime.rs

```diff
@@ -434,11 +434,11 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
             self.resolve_elided_lifetimes(slice::ref_slice(lifetime_ref));
             return;
         }
-        if lifetime_ref.name == keywords::StaticLifetime.name() {
+        if lifetime_ref.is_static() {
             self.insert_lifetime(lifetime_ref, Region::Static);
             return;
         }
-        self.resolve_lifetime_ref(lifetime_ref);
+        self.resolve_lifetime_ref(lifetime_ref, lifetime_ref);
```

Comment:
I would make the second parameter `Option<&hir::Lifetime>` and use `None` here to make it clear that the second argument won't get used.